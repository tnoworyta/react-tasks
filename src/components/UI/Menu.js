import React from 'react'
import { Link } from 'react-router-dom'

export default function UIMenu() {
  return (
    <React.Fragment>
    <div className="title">Some test app</div>
    <ul>
      <li>
        <Link to="/user/list">Users</Link>
      </li>
      <li>
        <Link to="/user/48342/edit">Edit some user</Link>
      </li>
      <li>
        <Link to="/product/list">Products</Link>
      </li>
    </ul>
    </React.Fragment>
  )
}
