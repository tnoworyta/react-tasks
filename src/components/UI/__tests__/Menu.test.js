import toJson from 'enzyme-to-json';

import Menu from '../Menu';

describe('<Menu />', () => {
  const defaultProps = { title: 'Some title' };
  const setup = buildSetup(Menu, defaultProps);

  test('renders component', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders with correct elements', () => {
    const wrapper = setup({ title: 'Some other title' });
    expect(wrapper.find('.title')).toHaveLength(1);
    expect(wrapper.find('li')).toHaveLength(3);
    expect(wrapper.find('.title').text()).toEqual('Some other title');
  });
});
