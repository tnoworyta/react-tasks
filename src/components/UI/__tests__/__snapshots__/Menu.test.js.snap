// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders component 1`] = `
<Fragment>
  <div
    className="title"
  >
    Some title
  </div>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
  <ul>
    <li>
      <Link
        to="/user/list"
      >
        Users
      </Link>
    </li>
    <li>
      <Link
        to="/user/48342/edit"
      >
        Edit some user
      </Link>
    </li>
    <li>
      <Link
        to="/product/list"
      >
        Products
      </Link>
    </li>
  </ul>
</Fragment>
`;
